modules/compute/src/main/java/org/apache/ignite/internal/compute/ComputeUtils.java

```diff
@@ -103,20 +110,69 @@ public static DeploymentUnitMsg toDeploymentUnitMsg(DeploymentUnit unit) {
                 .build();
     }
 
+    /**
+     * Extract compute job id from execute response.
+     *
+     * @param networkMessage Execution message response.
+     * @return Completable future with result.
+     */
+    public static CompletableFuture<UUID> jobIdFromExecuteResponse(NetworkMessage networkMessage) {
```

Comment:
I think it is better to move the exact type into the method signature. The casting will be explicit and won't cause unexpected exceptions.